libcaf_net/src/ip.cpp

```diff
@@ -106,6 +193,56 @@ std::vector<ip_address> resolve(string_view host) {
   return results;
 }
 
+std::vector<ip_address> resolve(ip_address host) {
+  return resolve(to_string(host));
+}
+
+std::vector<ip_address> local_addresses(string_view host) {
+  ip_address host_ip;
+  std::vector<ip_address> results;
+  if (host.empty()) {
+    for_each_adapter(
+      [&](string_view, ip_address ip) { results.push_back(ip); });
+  } else if (host == localhost) {
+    auto v6_local = ip_address{{0}, {0x1}};
+    auto v4_local = ip_address{make_ipv4_address(127, 0, 0, 1)};
+    for_each_adapter([&](string_view, ip_address ip) {
+      if (ip == v4_local || ip == v6_local)
+        results.push_back(ip);
+    });
+  } else if (auto err = parse(host, host_ip)) {
+    for_each_adapter([&](string_view iface, ip_address ip) {
+      if (iface == host)
+        results.push_back(ip);
+    });
+  } else {
+    return local_addresses(host_ip);
+  }
+  return results;
+}
+
+std::vector<ip_address> local_addresses(ip_address host) {
+  auto v6_any = ip_address{{0}, {0}};
+  auto v4_any = ip_address{make_ipv4_address(0, 0, 0, 0)};
+  // TODO: If is any addr, call resolve with PR #23.
```

Comment:
Are you planning to address this in this PR or separately? I'm fine with doing it later, but then please open an issue for the remaining work.

Reply:
Tried to addresses that but it seems I screwed up somewhere. I'll take a look at it tomorrow.

Reply:
Didn't think the last commit would fix the problem, but apparently it did.